source/extensions/filters/http/aws_lambda/aws_lambda_filter.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/http/filter.h"
+
+#include "extensions/common/aws/signer.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AwsLambdaFilter {
+
+class Arn {
+public:
+  Arn(absl::string_view partition, absl::string_view service, absl::string_view region,
+      absl::string_view account_id, absl::string_view resource_type,
+      absl::string_view function_name)
+      : partition_(partition), service_(service), region_(region), account_id_(account_id),
+        resource_type_(resource_type), function_name_(function_name) {}
+
+  const std::string& partition() const { return partition_; }
+  const std::string& service() const { return service_; }
+  const std::string& region() const { return region_; }
+  const std::string& accountId() const { return account_id_; }
+  const std::string& resourceType() const { return resource_type_; }
+  const std::string& functionName() const { return function_name_; }
+
+private:
+  std::string partition_;
+  std::string service_;
+  std::string region_;
+  std::string account_id_;
+  std::string resource_type_;
+  std::string function_name_; // resource_id
```

Comment:
nit: const. Same below/elsewhere where possible.

Reply:
I can't mark those members `const` because I need to update `arn_` member. FWIW, this class started as a simple POD with `const` members.